### PR TITLE
infra: upgrade frontend dependencies

### PR DESCRIPTION
**Pull request by stephanwlee**
This change upgrades our dependencies. Namely,
- angular
- bazel/angular
- rxjs
- NOT ngrx (ngrx10 causes karma test to fail)

With this change, our strict_deps check got much better and, in this PR, we fixed some and
cleaned up several typehacks that are no longer needed.

### CODE REVIEW

**Comment by stephanwlee** [comment]
Some part of the change here is fixed by #4378 to unblock the sync.

File: package.json [hunk 1]
```diff
@@ -28,20 +28,20 @@
...
+    "@bazel/ibazel": "^0.14.0",
+    "@bazel/jasmine": "^2.3.0",
+    "@bazel/karma": "^2.3.0",
+    "@bazel/protractor": "^2.3.0",
+    "@bazel/rollup": "^2.3.0",
+    "@bazel/terser": "^2.3.0",
+    "@bazel/typescript": "^2.3.0",
+    "@rollup/plugin-commonjs": "^16.0.0",
+    "@rollup/plugin-node-resolve": "^10.0.0",
+    "@types/chai": "^4.2.14",
```
Comment (psybuzz) [inline package.json:44]: Is `types/chai` still needed, now that our active tests were moved to jasmine after https://github.com/tensorflow/tensorboard/pull/4392 ?

**Review by psybuzz — COMMENTED** [review]
Code-wise, looks good, though locally I get some build errors:

Something about the tsconfig.json not liking lib: es2020, and

```
ERROR: .../tensorboard/webapp/widgets/line_chart_v2/lib/BUILD:7:14: output 'tensorboard/webapp/widgets/line_chart_v2/lib/internal_types.mjs' was not created
ERROR: .../tensorboard/webapp/widgets/line_chart_v2/lib/BUILD:7:14: output 'tensorboard/webapp/widgets/line_chart_v2/lib/scale_types.mjs' was not created
ERROR: .../tensorboard/webapp/widgets/line_chart_v2/lib/BUILD:7:14: not all outputs were created or valid
ERROR: .../tensorboard/webapp/app_routing/BUILD:52:14: output 'tensorboard/webapp/app_routing/types.mjs' was not created
ERROR: .../tensorboard/webapp/third_party/BUILD:7:14: output 'tensorboard/webapp/third_party/d3.mjs' was not created
```

**Comment by stephanwlee** [comment]
> though locally I get some build errors:
Something about the tsconfig.json not liking lib: es2020, and

What command did you use? `bazel clean --expunge` worked okay for me. Also the CI seems to chugging along just fine.

**Review by psybuzz — APPROVED** [review]

**Pull request merged**